### ISSUE TITLE
[tests] Ray nightly tests fail due to OOMs

### ISSUE THREAD

**Issue by tgaddair** [ludwig-ai/ludwig#2451]
Ray nightly tests are failing for the Horovod + Hyperopt tests due to OOMs. These tests are the longest running and most resource intensive (7 CPUs per Ray cluster), so it is not surprising that if any tests would fail due to resource issues, these would be the ones.

```
(run_experiment_trial pid=23448) 
Evaluation test :   0%|          | 0/1 [00:00<?, ?it/s]
(run_experiment_trial pid=23448) 
Evaluation test : 100%|██████████| 1/1 [00:00<00:00,  9.14it/s]
Evaluation test : 100%|██████████| 1/1 [00:00<00:00,  8.19it/s]
(PipelineSplitExecutorCoordinator pid=23995) 
Stage 0: : 6it [00:22,  1.72s/it]
(raylet) [2022-09-04 16:20:51,029 E 17085 17125] (raylet) node_manager.cc:2963: System memory low at node with IP 10.1.0.153. Used memory (6.13GB) / total capacity (6.78GB) (0.904609) exceeds threshold 0.9, killing latest task with name _map_block_nosplit() and task ID ba3bc8179d50d2f3d8870fc35f6238456f07268601000000 to avoid running out of memory.
(raylet) This may indicate a memory leak in a task or actor, or that too many tasks are running in parallel.
(raylet) To find the highest memory consumers, use `ray logs raylet.out -ip 10.1.0.153`.
(raylet) Consider provisioning more memory on this node or reducing task parallelism by requesting more CPUs per task. To adjust the eviction threshold, set the environment variable `RAY_memory_usage_threshold_fraction` when starting Ray. To disable worker eviction, set the environment variable `RAY_memory_monitor_interval_ms` to zero.
```

Note the issues do not repro locally, where we have more memory.

https://github.com/ludwig-ai/ludwig/runs/8177672264?check_suite_focus=true#step:10:7153

**Comment by ShreyaR** [comment]
Possible fix in #2599.

**Comment by arnavgarg1** [comment]
Fixed by https://github.com/ludwig-ai/ludwig/pull/2599
